### PR TITLE
[stable/jenkins] Add runAsUser and runAsGroup to podTemplate

### PR DESCRIPTION
**Pull request by janekbaraniewski**
No

#### What this PR does / why we need it:
Adds ability to configure user/group for Jenkins agent containers. Currently it's impossible to do using Jenking Configuration as a Code because of those missing values, causing failure of custom agent sidecar containers with different default user.

#### Special notes for your reviewer:
@lachie83 Hey, first time making a PR to charts repo so not really sure how it works, should I tag all maintainers here?

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/jenkins/templates/_helpers.tpl [hunk 1]
```diff
@@ -190,6 +190,8 @@ Returns kubernetes pod template configuration as code
     resourceLimitMemory: {{.Values.agent.resources.limits.memory}}
     resourceRequestCpu: {{.Values.agent.resources.requests.cpu}}
     resourceRequestMemory: {{.Values.agent.resources.requests.memory}}
+    runAsUser: {{ .Values.agent.runAsUser }}
+    runAsGroup: {{ .Values.agent.runAsGroup }}
```
Comment (wmcdona89) [inline stable/jenkins/templates/_helpers.tpl:196]: could you explain your use case in more detail? these fields can be configured for both the pod and the container and I'm wondering if there's a need to support both. if so, the naming will need to account for this.

perhaps...
```
agent.runAsUserContainer
agent.runAsGroupContainer
agent.runAsUserPod
agent.runAsGroupPod
```

**Review by wmcdona89 — CHANGES_REQUESTED** [review]
@janekbaraniewski thanks for the PR!

please rebase and also update `values.yaml` and the CHANGELOG

**Comment by torstenwalter** [comment]
@wmcdona89 Isn't it already possible to configure it using podTemplate?
In that case it could be better to add an example rather than introducing yet another field.

**Comment by janekbaraniewski** [comment]
@torstenwalter that's true, it is possible to configure using `agent.podTemplates`. However, this doesn't suit my usecase that well. 

I need to define common "base" configuration that then can have specific fields overwritten depending on environment. Since `agent.podTemplates` is essentially a string I cannot simply overwrite for example `image` attribute without redefining whole structure again. I could introduce some rendering mechanism but that introduces additional complexity to my configuration and I would like to avoid it if possible.

I also can't see a reason why it should be possible to do it using `podTemplates` but not `additionalAgents`. From a user perspective I think that's misleading since both of them claim to do the same thing - `configure additional kubernetes pod templates.`

File: stable/jenkins/templates/_helpers.tpl (hunk 1 above)
Comment (janekbaraniewski) [reply stable/jenkins/templates/_helpers.tpl:196]: Hey, I'm using custom agent sidecar image with some custom default user `example` which doesn't have proper permissions to use mounted jenkins workspace that's  managed by `jnlp` container. 
So in my case I only need to modify them at container level and I'm not sure if there is a need to set it at pod level. Only example I can think of is when you want to also change user in `jnlp` sidecar that's added to agent pod when you overwrite `sideContainerName` and set it to one value for all containers.

Comment (wmcdona89) [reply stable/jenkins/templates/_helpers.tpl:196]: Yeah, setting these options at the container level does seem like the more common use case.

**Comment by wmcdona89** [comment]
@torstenwalter lgtm

**Pull request merged**